lib/trino-hive-formats/src/main/java/io/trino/hive/formats/avro/NativeLogicalTypesAvroTypeManager.java

```diff
@@ -338,4 +449,110 @@ public static long fromBigEndian(byte[] bytes)
         }
         return res;
     }
+
+    public static byte[] fitBigEndianValueToByteArraySize(long value, int byteSize)
+    {
+        return fitBigEndianValueToByteArraySize(Longs.toByteArray(value), byteSize);
+    }
+
+    public static byte[] fitBigEndianValueToByteArraySize(Int128 value, int byteSize)
+    {
+        return fitBigEndianValueToByteArraySize(value.toBigEndianBytes(), byteSize);
+    }
+
+    /**
+     * Will resize big endian bytes to a desired array length while preserving the represented value.
+     * Will throw an arithmetic exception if it is not possible
+     */
+    public static byte[] fitBigEndianValueToByteArraySize(byte[] value, int byteSize)
+    {
+        if (value.length == byteSize) {
+            return value;
+        }
+        else if (value.length < byteSize) {
+            return padBigEndianToSize(value, byteSize);
+        }
+        else if (canBigEndianValueBeRepresentedByByteSize(value, byteSize)) {
+            byte[] dest = new byte[byteSize];
+            System.arraycopy(value, value.length - byteSize, dest, 0, byteSize);
+            return dest;
+        }
+        throw new ArithmeticException("Can't resize big endian bytes %s to size %s".formatted(Arrays.toString(value), byteSize));
+    }
+
+    public static boolean canBigEndianValueBeRepresentedByByteSize(byte[] bigEndianValue, int byteSize)
+    {
+        // pre-req 1
+        // can't represent number with 0 bytes
+        if (byteSize <= 0) {
```

Comment:
These first two checks are redundant, since they are already tested above, before this method is called. Let's change this method to be private and make the tests verify that `fitBigEndianValueToByteArraySize()` throws when expected.

Reply:
believe this first condition is still needed. Removing second and restricting method to smaller valueLengths that targetByteSize.